tests/runner/src/main.rs

```diff
@@ -1,41 +1,50 @@
-mod setup;
-mod cli;
-mod tests;
-mod environment;
-mod tls;
-mod test_runner;
-
-pub use cli::*;
-pub use tls::*;
-
-const VERSION: &str = include_str!("../../../VERSION");
-
-fn main() {
-    use fluvio_future::task::run_block_on;
-    use setup::Setup;
-    use test_runner::TestRunner;
-
-    println!("Start running fluvio test runner");
-    fluvio_future::subscriber::init_logger();
-
-    let option = TestOption::parse_cli_or_exit();
-
-    let test_runner = TestRunner::new(option.clone());
-
+#[tokio::main]
+async fn main() {
     // catch panic in the spawn
     std::panic::set_hook(Box::new(|panic_info| {
         eprintln!("panic {}", panic_info);
         std::process::exit(-1);
     }));
 
-    run_block_on(async move {
-        if option.setup() {
-            let mut setup = Setup::new(option);
-            setup.setup().await;
-        } else {
-            println!("no setup");
-        }
+    use fluvio_test_util::test_meta::TestOption;
+    use fluvio::Fluvio;
+
+    println!("Start running fluvio test runner");
+    fluvio_future::subscriber::init_logger();
+
+    let option = TestOption::parse_cli_or_exit();
 
-        test_runner.run_test().await;
-    });
+    // Deploy a cluster
+    let fluvio_client = if option.skip_cluster_start() {
+        println!("skipping cluster start");
+        // Connect to cluster in profile
+        Fluvio::connect().await.unwrap()
+    } else {
+        let mut test_cluster = fluvio_test_util::setup::TestCluster::new(option.clone());
+        test_cluster.start().await.unwrap()
+    };
+
+    // TODO: Build this with Test Runner
+    if let Some(test_name) = &option.test_name {
+        match test_name.as_str() {
```

Comment:
this doesn't scale, we can't hard code tests every time we add test

Reply:
Understood. I was landing short so the change would not be as large and risky for this PR because we use the smoke tests.

We still only support smoke tests running successfully, and nothing more. But we have an "improved but still static" interface to add new tests. A more scaled approach will come in the next PR if that is ok with you?